cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -211,14 +209,28 @@ public static LoggingContext getLoggingContext(String namespaceId, String applic
 
   public static Filter createFilter(LoggingContext loggingContext) {
     if (loggingContext instanceof ServiceLoggingContext) {
-      String systemId = loggingContext.getSystemTagsMap().get(ServiceLoggingContext.TAG_NAMESPACE_ID).getValue();
+      LoggingContext.SystemTag namespaceOrSystemID = getNamespaceOrSystemID(loggingContext.getSystemTagsMap());
+      if (namespaceOrSystemID == null) {
+        throw new IllegalArgumentException("No namespace or system id present");
+      }
+      String systemId = namespaceOrSystemID.getValue();
```

Comment:
Seems confusing after calling `getValue()`, the `namespaceOrSystemID` becomes `systemId`. Better call them `systemTag` and `systemId`.